enclave-runtime/src/lib.rs

```diff
@@ -622,25 +632,33 @@ where
 	let mut validator = LightClientSeal::<PB>::unseal()?;
 	let mut nonce = NONCE.write().expect("Encountered poisoned NONCE lock");
 
-	sync_blocks_on_light_client(blocks_to_sync, &mut validator, &OcallApi, &mut *nonce)?;
+	sync_blocks_on_light_client(
+		blocks_to_sync,
+		&mut validator,
+		&OcallApi,
+		&GlobalFileStateHandler,
+		&mut *nonce,
+	)?;
```

Comment:
In all the places where we need to somehow handle access to the STF state, we now need an instance of the `HandleState` trait. In 'regular', i.e. non-testing code, this will currently be the `GlobalFileStateHandler`.